crates/relayer/src/chain/axon.rs

```diff
@@ -687,6 +701,83 @@ impl ChainEndpoint for AxonChain {
     ) -> Result<Vec<CrossChainQueryResponse>, Error> {
         todo!()
     }
+
+    fn build_connection_proofs_and_client_state(
+        &self,
+        message_type: ConnectionMsgType,
+        connection_id: &ConnectionId,
+        client_id: &ClientId,
+        height: Height,
+    ) -> Result<(Option<AnyClientState>, Proofs), Error> {
+        let message_type = match message_type {
+            ConnectionMsgType::OpenTry => connection::State::Init,
+            ConnectionMsgType::OpenAck => connection::State::TryOpen,
+            ConnectionMsgType::OpenConfirm => connection::State::Open,
+        };
+        let key = (connection_id.clone(), message_type);
+        let tx_hash = self.conn_tx_hash.get(&key).unwrap();
```

Comment:
do not use unwrap() here, emit error if no key in the mapping

Reply:
The `tx_hash` is saved by our flow and if we can't get it then it means something expected happen, so I unwrap it. Should I return `err` here?